### ISSUE TITLE
[BUG]:  render projects only with `access.status === 'active'`

### ISSUE THREAD

**Issue by dahal** [envless/envless#395]
### Describe the bug

We are currently rendering all projects regardless of `access.status` on `/projects` endpoint. 

### Steps to reproduce

1. Setup project
2. Run seed and login
3. Go to `/projects`
4. You will see all projects you have access to, regardless of your `access.status`

### Expected behavior

We need to do two things:-

1. Render projects on `/projects` with `access.status === 'active'`
2. Render 404 when users visit project's children endpoints (eg: `projects/{slug}/branches` if their `access.status` is not `active`

**Comment by ttebify** [comment]
I am in for this

**Comment by ttebify** [comment]
Hi @dahal, do we have to tell them the reason why they cannot access the protected pages? For example, when a user's account is locked, they are taken to the locked page. What do you think?

**Comment by dahal** [comment]
@ttebify here are few scenarios

1. Locked users, should not be able to login, or go to any of the `/project` pages directly (I think we handle this and render `locked.tsx`) already. Worth double checking.
2. If a user has access to project, and the access.status is
    * `inactive or pending` - they should not have access to any of the `/projects` children pages, we should show them 404
    * `active` - they can access the page just fine, based on their access.role (which should be handled differently.


TLDR - we don't have to custom render the reasoning why they cannot access the page.
